src/Processors/Worker.php

```diff
@@ -37,36 +33,23 @@ public function __construct(
      */
     public function process(MessageInterface $message, Queue $queue): void
     {
-        $event = ExecEvent::before($message->getId(), $message->getJob(), $message->getTtr(), $message->getAttempt());
-
-        $title = $this->formatter->getExecTitle($event);
-        $this->logger->info("$title is started.");
+        /** @var BeforeExecutionInterface $event */
+        $event = $this->factory->create(BeforeExecutionInterface::class, [$queue, $message]);
```

Comment:
What's the use case to configure event classes via factory?

Reply:
It's a good way to make your own implementation of these events specifically for your project.

Reply:
Wouldn't it be better to implement a factory like in widgets than to inject the entire container into the class?

Reply:
And what are the pros of doing so?

Reply:
That I could call the factory from anywhere you need, without the need to inject it to the class.

Reply:
Still didn't get you :( Could you provide an example of "calling the factory from anywhere"?

Reply:
> your own implementation of these events specifically for your project

Could you provide an example when you need so?

Reply:
- Event can provide some extended functionality, i.e. smthng like `isPropagationStopped` for the current specific case (`isSomethingHappened`).
- You can extend default event class with your own one in order to subscribe just for this implementation, so in this case the event will be cought by all the default listeners and this one without any `if` inside (viable for monolith projects with several queues):

```php
class OwnEvent extends DefaultEvent {}

$listener = fn(OwnEvent::class) => null;

$listener->subscribe($listener);
```

Reply:
Decided to use `new class()` instead of factory in a private discussion with @samdark.